http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpJupyterDAO.scala

```diff
@@ -1,54 +1,52 @@
 package org.broadinstitute.dsde.workbench.leonardo.dao
 
-import akka.actor.ActorSystem
-import akka.http.scaladsl.Http
-import akka.http.scaladsl.model._
-import akka.http.scaladsl.unmarshalling.Unmarshal
-import akka.stream.ActorMaterializer
+import cats.effect.{Concurrent, ContextShift, Timer}
+import cats.implicits._
 import com.typesafe.scalalogging.LazyLogging
 import io.circe.Decoder
-import io.circe.parser.parse
 import org.broadinstitute.dsde.workbench.leonardo.dao.ExecutionState.{Idle, OtherState}
 import org.broadinstitute.dsde.workbench.leonardo.dao.HttpJupyterDAO._
 import org.broadinstitute.dsde.workbench.leonardo.dns.ClusterDnsCache
 import org.broadinstitute.dsde.workbench.leonardo.dns.ClusterDnsCache._
 import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.client.Client
+import org.http4s.{Method, Request, Uri}
 
-import scala.concurrent.{ExecutionContext, Future}
-
-class HttpJupyterDAO(val clusterDnsCache: ClusterDnsCache)(implicit system: ActorSystem,
-                                                           executionContext: ExecutionContext,
-                                                           materializer: ActorMaterializer)
-    extends JupyterDAO
+class HttpJupyterDAO[F[_]: Timer: ContextShift: Concurrent](val clusterDnsCache: ClusterDnsCache[F], client: Client[F])
+    extends JupyterDAO[F]
     with LazyLogging {
-
-  val http = Http(system)
-
-  def isProxyAvailable(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] =
-    Proxy.getTargetHost(clusterDnsCache, googleProject, clusterName) flatMap {
+  def isProxyAvailable(googleProject: GoogleProject, clusterName: ClusterName): F[Boolean] =
+    Proxy.getTargetHost[F](clusterDnsCache, googleProject, clusterName) flatMap {
       case HostReady(targetHost) =>
-        val statusUri = Uri(s"https://${targetHost.toString}/notebooks/$googleProject/$clusterName/api/status")
-        http.singleRequest(HttpRequest(uri = statusUri)) map { response =>
-          response.status.isSuccess
-        }
-      case _ => Future.successful(false)
+        client.successful(
+          Request[F](
+            method = Method.GET,
+            uri = Uri.unsafeFromString(
+              s"https://${targetHost.toString}/notebooks/$googleProject/$clusterName/api/status"
+            )
+          )
+        )
+      case _ => Concurrent[F].pure(false)
     }
 
-  def isAllKernalsIdle(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] =
+  def isAllKernalsIdle(googleProject: GoogleProject, clusterName: ClusterName): F[Boolean] =
```

Comment:
This was here before, but `kernel` is misspelled :)

Reply:
it was probably me who introduced the typo :) will update